text/0750-feature-flag-advisor.md

```diff
@@ -0,0 +1,343 @@
+
+```
+> cdk flags --set-recommended --all
+    Feature Flag                              Recommended Value            User Value
+  * @aws-cdk/...                              true                         false
+  * @aws-cdk/...                              true                         false
+  * @aws-cdk/...                              true                         - <unset>
+  Setting all feature flags to the recommended values...
+
+  Resynthesizing... 
+
+  Here is the difference:
+  - S3 Bucket:
+  -    Properties:
+  -      `BucketEncryption``:`` ``None`
+  + S3 Bucket:
+  +    Properties:
+  +      BucketEncryption: 
+  +        `ServerSideEncryptionConfiguration``:`
+  +          ...
+  ```
+
+#### I want to change my unconfigured flags to the recommended states
+
+To change your unconfigured feature flags to our recommended states, you can run `cdk flags --set-recommended --unconfigured`.
+
+```
+> cdk flags --set-recommended --unconfigured
+    Feature Flag                              Recommended Value            User Value
+  * @aws-cdk/...                              true                         - <unset>
+  Setting all feature flags to the recommended values...
+
+  Resynthesizing... 
+  
+  Here is the difference:
+  - S3 Bucket:
+  -    Properties:
+  -      `BucketEncryption``:`` ``None`
+  + S3 Bucket:
+  +    Properties:
+  +      BucketEncryption: 
+  +        `ServerSideEncryptionConfiguration``:`
+  +          ...
+  
+```
+
+#### I only want to change one feature flag to its recommended state
+
+To simply modify one feature flag to our recommended state, you can run `cdk flags --set-recommended --flagName="#FLAGNAME#"`.
+
+```
+> cdk flags --set-recommended --flagName="#FLAGNAME#"
+    Feature Flag                              Recommended Value            User Value
+  * @aws-cdk/...                              true                         false
+  Setting all feature flags to the recommended values...
+
+  Resynthesizing... 
+  
+  Here is the difference:
+  - S3 Bucket:
+  -    Properties:
+  -      `BucketEncryption``:`` ``None`
+  + S3 Bucket:
+  +    Properties:
+  +      BucketEncryption: 
+  +        `ServerSideEncryptionConfiguration``:`
+  +          ...
+  
+```
+
+#### I want to modify one feature flag to a different value
+
+To modify one feature flag, first we view its information.
+
+```
+> cdk flags "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021"
+    Description: Enable this feature flag to have cloudfront distributions use the security policy TLSv1.2_2021 by default.
+    Recommended Value: true
+    User Value: true
+```
+
+Additionally, to modify the state, you can run `cdk flags --set --flagName="#FLAGNAME#"`. This will return a list of values for you to choose from.
+
+```
+> cdk flags "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021" --set
+  What value would you like to change "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021" to?
```

Comment:
Is there a non-interactive way I can apply these changes? Even though the feature will likely mainly used by humans, we try to offer non-interactive versions to everything we do, so that people can create scripts with these capabilities.